feat(clusterissuer): support desec via webhook

ruzko opened the pull request:
**Description**
The Clusterissuer chart supports few issuers,
only the ones integrated in Cert-manager.
Cert-manager has since v0.8 (see [link](https://cert-manager.io/docs/releases/release-notes/release-notes-0.8/#webhook-based-acme-dns01-solver)
) limited the number of ACME providers
they support within their codebase,
and opened up for external webhooks.
[This webhook](https://github.com/irreleph4nt/cert-manager-webhook-desec-http) implements support for the
non-profit, open source DNS provider deSEC
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
⚒️ Fixes  # <!--(issue)-->
partially #17055 
**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
My Trunas Scale install is semi-borked, and I don't have a cluster on my laptop.
Ergo, this hasn't been tested.

<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/truecharts/charts?pullRequest=18005) <br/>All committers have signed the CLA.

ruzko committed:
feat(clusterissuer): support desec via webhook

The Clusterissuer chart supports few issuers,
only the ones integrated in Cert-manager.
Cert-manager has since v0.8 (see \
https://cert-manager.io/docs/releases/release-notes/\
release-notes-0.8/#webhook-based-acme-dns01-solver)
limited the number of ACME providers they support
within their codebase, and opened up for
external webhooks.
This webhook implements support for the
non-profit, open source DNS provider deSEC.

Crow-Control commented:
At this point in time, i'm going to VETO this.
As we currently only load external helm-charts in a few select cases and without a policy on it's a "no unless explicitly maintainer approved".

---

It's not a no forever and we do want this implemented, but we do want to carefully look into how we want to do it correctly.
